src/main/java/io/cdap/cdap/events/process/pubsub/PubSubEventWriter.java

```diff
@@ -0,0 +1,206 @@
+     *
+     * @param builder   Pub/Sub builder where was already initialized with the topic name, and project
+     * @param proxyHost Proxy host where configure the use of the proxy
+     * @param proxyPort Proxy port where configure the use of the proxy
+     * @throws NumberFormatException
+     */
+    private void configureChannelProxy(Publisher.Builder builder, String proxyHost, String proxyPort)
+            throws NumberFormatException {
+        SocketAddress proxySocketAddress = new InetSocketAddress(proxyHost, Integer.parseInt(proxyPort));
+        builder.setChannelProvider(
+                InstantiatingGrpcChannelProvider.newBuilder()
+                        .setChannelConfigurator(managedChannelBuilder -> managedChannelBuilder.proxyDetector(
+                                new ProxyDetector() {
+                                    @Nullable
+                                    @Override
+                                    public ProxiedSocketAddress proxyFor(SocketAddress socketAddress) {
+                                        if (socketAddress == null) {
+                                            return null;
+                                        }
+                                        return HttpConnectProxiedSocketAddress.newBuilder()
+                                                .setTargetAddress((InetSocketAddress) socketAddress)
+                                                .setProxyAddress(proxySocketAddress)
+                                                .build();
+                                    }
+                                }))
+                        .build()
+        );
+    }
+
+    @Override
+    public void publishEvent(Event event) {
+        if (publisher == null) {
+            logger.info("Publisher is not already initialized");
+            return;
+        }
+        logger.info("Publishing event");
+        String stringEvent = GSON.toJson(event);
+        ByteString data = ByteString.copyFromUtf8(stringEvent);
+        try {
+            PubsubMessage pubsubMessage = PubsubMessage.newBuilder()
+                    .setData(data)
+                    .build();
+            ApiFuture<String> future = this.publisher.publish(pubsubMessage);
+            ApiFutures.addCallback(
+                    future,
+                    new ApiFutureCallback<String>() {
+
+                        @Override
+                        public void onFailure(Throwable e) {
+                            logger.error("Error publishing message : " + e.getMessage());
+                        }
+
+                        @Override
+                        public void onSuccess(String messageId) {
+                            logger.info("Published message ID: " + messageId);
+                        }
+                    },
+                    MoreExecutors.directExecutor());
+            int retries = 0;
+            while (!future.isDone() && retries <= 10) {
```

Comment:
As discussed in a previous review, is it required to wait for future to be done here?